### PR TITLE
Refactor CMainMenu and CModManager

### PR DESCRIPTION
**Pull request by Moris-Onz**
Refactor CMainMenu and CModManager

### CODE REVIEW

File: Client/core/CMainMenu.cpp [hunk 1]
```diff
@@ -305,45 +305,6 @@ CMainMenu::CMainMenu(CGUI* pManager)
...
-
-    m_pFeatureBranchAlertLabel.reset(reinterpret_cast<CGUILabel*>(m_pManager->CreateLabel(m_pFeatureBranchAlertImage.get())));
-    m_pFeatureBranchAlertLabel->SetText(
-        _("You are using a feature-branch build! This is a test build only which cannot be used to connect to public servers!"));
-    m_pFeatureBranchAlertLabel->SetPosition({0.0f, 0.0f}, false);
-    m_pFeatureBranchAlertLabel->SetSize({ScreenSize.fX, 35.0f});
-    m_pFeatureBranchAlertLabel->SetFont("clear-normal");
-    m_pFeatureBranchAlertLabel->SetHorizontalAlign(CGUI_ALIGN_HORIZONTALCENTER);
-    m_pFeatureBranchAlertLabel->SetVerticalAlign(CGUI_ALIGN_VERTICALCENTER);
-#endif
```
Comment (botder) [inline Client/core/CMainMenu.cpp:328]: Why was this functionality removed?

File: Client/core/CMainMenu.cpp [hunk 2]
```diff
@@ -382,18 +343,17 @@ void CMainMenu::SetMenuVerticalPosition(int iPosY)
...
-        m_pHoveredItem = NULL;
+        m_pHoveredItem = nullptr;
     }
 
     float fFirstItemSize = m_menuItems.front()->image->GetSize(false).fY;
     int   iMoveY = iPosY - m_menuItems.front()->image->GetPosition(false).fY - fFirstItemSize * 0.5f;
 
-    std::deque<sMenuItem*>::iterator it = m_menuItems.begin();
-    for (it; it != m_menuItems.end(); it++)
+    for (auto it : m_menuItems)
```
Comment (botder) [inline Client/core/CMainMenu.cpp:352]: Give it the actual type and a proper name.

File: Client/core/CMainMenu.cpp [hunk 3]
```diff
@@ -429,6 +390,12 @@ void CMainMenu::Update()
         m_bHideGame = false;
     }
 
+    // Make sure frame rate limit gets applied
+    if (m_bIsIngame)
+        CCore::GetSingleton().EnsureFrameRateLimitApplied(); // Catch missed frames
+    else
+        CCore::GetSingleton().ApplyFrameRateLimit(88);       // Limit when not connected
+
```
Comment (botder) [inline Client/core/CMainMenu.cpp:398]: Why was this logic moved and why was the condition changed `if (m_pClientBase)`?

File: Client/core/CModManager.cpp [hunk 4]
```diff
@@ -19,13 +19,13 @@
...
-CModManager* CSingleton<CModManager>::m_pSingleton = NULL;
+CModManager* CSingleton<CModManager>::m_pSingleton = nullptr;
 
 CModManager::CModManager()
 {
     // Init
-    m_hClientDLL = NULL;
-    m_pClientBase = NULL;
+    m_hClientDLL = nullptr;
+    m_pClientBase = nullptr;
```
Comment (botder) [inline Client/core/CModManager.cpp:28]: Initialize the member variables in the header instead.

File: Client/core/CModManager.cpp [hunk 5]
```diff
@@ -88,7 +88,7 @@ void CModManager::ClearRequest()
 
 bool CModManager::IsLoaded()
 {
-    return (m_hClientDLL != NULL);
+    return (m_hClientDLL != nullptr);
 }
```
Comment (botder) [inline Client/core/CModManager.cpp:92]: This logic can be moved to the header file.

File: Client/core/CModManager.cpp [hunk 6]
```diff
@@ -213,9 +213,9 @@ void CModManager::Unload()
         CLocalGUI::GetSingleton().GetDebugView()->Clear();
         CLocalGUI::GetSingleton().SetDebugViewVisible(false);
 
-        // NULL the message processor and the unhandled command handler
-        CCore::GetSingleton().SetClientMessageProcessor(NULL);
-        CCore::GetSingleton().GetCommands()->SetExecuteHandler(NULL);
+        // Nullptr the message processor and the unhandled command handler
```
Comment (botder) [inline Client/core/CModManager.cpp:216]:
```suggestion
        // Reset the message processor and the unhandled command handler
```

File: Client/core/CModManager.cpp [hunk 7]
```diff
@@ -347,7 +341,7 @@ void CModManager::InitializeModList(const char* szModFolderPath)
         VerifyAndAddEntry(szModFolderPath, ToUTF8(FindData.cFileName));
 
         // Search until there aren't any files left
-        while (FindNextFileW(hFind, &FindData) == TRUE)
+        while (FindNextFileW(hFind, &FindData) == true)
```
Comment (botder) [inline Client/core/CModManager.cpp:344]: `FindNextFileW` does not return a C++ `bool` but a Win32 `BOOL`, which is different.

**Review by botder — CHANGES_REQUESTED** [review]

File: Client/core/CMainMenu.cpp (hunk 2 above)
Comment (Moris-Onz) [reply Client/core/CMainMenu.cpp:352]: ?

**Pull request closed without merging**